.github/workflows/test.yml

```diff
@@ -2,32 +2,50 @@ name: Test
 
 on:
   push:
-    branches: ['*']
+    branches: [main]
   pull_request:
-    branches: ['*']
+    branches: [main]
 
 jobs:
   bud:
     name: Test
     runs-on: ${{ matrix.os }}
     strategy:
+      fail-fast: false
       matrix:
         os: [ubuntu-latest, macos-latest]
+        go: ['1.16', '1.17', '1.18']
+        node: ['16']
+        include:
+          - os: ubuntu-latest
+            go: 1.18
+            node: 14
+          - os: ubuntu-latest
+            go: 1.18
+            node: 18
+          - os: macos-latest
+            go: 1.18
+            node: 14
+          - os: macos-latest
+            go: 1.18
+            node: 18
 
     steps:
       - name: Check out bud
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
 
       - name: Set up Go
-        uses: actions/setup-go@v2
+        uses: actions/setup-go@v3
         id: go
         with:
-          go-version: 1.18.0
+          go-version: ${{ matrix.go }}
+          cache: true
+          cache-dependency-path: go.sum
```

Comment:
caching is built-in since v3, so we can omit the previous workaround

Reply:
Just noticed that it was part of the recent release of this action while upgrading and was hoping it would work this way straight out of the box. I will have a deeper look on this and play around a little bit with the action. 

If you prefer to merge the matrix testing beforehand, I could easily revert the previous caching mechanism and provide an upcoming PR once I'm confident how to use it via the action itself.

Note: A first look on the action itself seems promising (see: https://github.com/actions/setup-go/blob/b22fbbc2921299758641fab08929b4ac52b32923/src/package-managers.ts#L13) as at least the same directories are used ;)

Reply:
Spawned up a little [playground](https://github.com/wheinze/setup-go-playground/actions) to check the action and actually it seems to work fine for me.

So I would recommend that you give it a try and grant the second actions run to check if the cache will be restored ;) Or do I miss something?

Reply:
I think it's fine to merge this as is. 

I didn't spend a lot of time testing or verifying CI yet, so I'm not even sure the existing setup is doing the correct caching. 

Thanks for testing this in a playground!